### PR TITLE
Added E2E test of ODH deployment

### PR DESCRIPTION
**Pull request by RobGeada**
See [the e2e readme](https://github.com/RobGeada/trustyai-explainability/tree/e2e_test/e2e_tests) for more info

Many thanks for submitting your Pull Request :heart:!

Please make sure that your PR meets the following requirements:

- [ ] You have read the [contributors guide](https://github.com/trustyai-explainability/trustyai-explainability/blob/main/CONTRIBUTING.md)
- [ ] Pull Request title is properly formatted: `FAI-XYZ Subject`
- [ ] Pull Request title contains the target branch if not targeting main: `[0.3.x] FAI-XYZ Subject`
- [ ] Pull Request contains link to the JIRA issue
- [ ] Pull Request contains link to any dependent or related Pull Request
- [ ] Pull Request contains description of the issue
- [ ] Pull Request does not include fixes for issues other than the main ticket

### CODE REVIEW

File: e2e_tests/odh-deployment-e2e.sh [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+fi
+
+# grab + run data generator  ===========================================================================================
+[ -d "logger" ] && rm -Rf logger
+cp -r ../explainability-service/demo/logger logger
+TRUSTY_ROUTE=$(oc get route/trustyai-service-route --template={{.spec.host}})
+cd logger
+
+sed "s/sleep(random.randint(1, 3))/sleep(0.01)/" partial.py > partial-fast-gen.py
+SERVICE_ENDPOINT=http://$TRUSTY_ROUTE/consumer/kserve/v2 python3 partial-fast-gen.py & sleep 60 ; kill $!
```
Comment (ruivieira) [inline e2e_tests/odh-deployment-e2e.sh]: Just a note that the `partial-fast-gen.py` is not included in this PR.

Also, (if based on `partial.py` it depends on the `requests` package that might need to be installed.

File: e2e_tests/odh-deployment-e2e.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+          "type": "DOUBLE",
+          "value": 0.0
+      },
+      "unprivilegedAttribute": {
+          "type": "DOUBLE",
+          "value": 1.0
+      }
+  }'
+
+
```
Comment (ruivieira) [inline e2e_tests/odh-deployment-e2e.sh:133]: (Time permitting) a GET request to `http://$TRUSTY_ROUTE/metrics/spd/requests` should show that the metric request is indeed registered.

File: e2e_tests/odh-deployment-e2e.sh [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+# print out prometheus route ===========================================================================================
+echo "Go to https://$(oc get route odh-model-monitoring --template={{.spec.host}}) and see if 'trusty_spd' is present"
+echo "If so, e2e is successful!"
+
+
```
Comment (ruivieira) [inline e2e_tests/odh-deployment-e2e.sh:140]: (Time permitting) a DELETE request to 
```
curl --location --request DELETE 'http://$TRUSTY_ROUTE/metrics/spd/request' \
--header 'Content-Type: application/json' \
--data '{
    "requestId": "$REQUEST_UUID"
}'
```
and subsequent GET request to `http://$TRUSTY_ROUTE/metrics/spd/requests` should show that the metric request was deleted and Prometheus is no longer updating it.

Comment (ruivieira) [reply e2e_tests/odh-deployment-e2e.sh:140]: (Time permitting) A request to `curl --location 'http://$TRUSTY_ROUTE/info'` should show dataset schemas, number of observations and registered requests.

**Review by ruivieira — APPROVED** [review]

File: e2e_tests/model.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  name: example-sklearn-isvc
+  annotations:
+    serving.kserve.io/deploymentMode: ModelMesh
+spec:
+  predictor:
+    model:
+      modelFormat:
+        name: sklearn
+      runtime: mlserver-0.x
+      storageUri: "https://github.com/ruivieira/kserve-trustyai-explainer/blob/predictor-model/predictor/model.joblib?raw=true"
```
Comment (danielezonca) [inline e2e_tests/model.yaml:13]: We should move this into the repo or in general use a [permanent link](https://github.com/ruivieira/kserve-trustyai-explainer/blob/55ef1fa76379ddad543dac2d130b5a4644449f02/predictor/model.joblib?raw=true) to be safer

File: e2e_tests/README.md [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+1) Select a particular fork of odh-manifests, e.g. https://github.com/$(REPO_OWNER)/odh-manifests
+2) Choose the branch to test `$BRANCH_NAME`
+3) Run the script, pointing at that fork and branch:
+
+`./odh-deployment-e2e $REPO_OWNER $BRANCH_NAME`
+
+
+## What this does (loosely)
+1) Cleans your cluster of previous e2e tests
+2) Deploys a minimal ODH install (currently uses PR versions of ModelMesh-serving and ModelMesh, see [here](https://github.com/RobGeada/odh-manifests/blob/trustyai-pr-images/model-mesh/base/params.env) for details)
```
Comment (danielezonca) [inline e2e_tests/README.md:26]: We should update this link with opendatahup repo

File: e2e_tests/odh-minimal.yaml [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+      - name: deployment-namespace
+        value: trustyai-e2e
+      repoRef:
+        name: manifests
+        path: modelmesh-monitoring
+    name: modelmesh-monitoring
+  repos:
+  - name: trustyai-pr-images-manifests
+    #uri: https://github.com/opendatahub-io/odh-manifests/tarball/master
+    uri: https://github.com/robgeada/odh-manifests/tarball/trustyai-pr-images
```
Comment (danielezonca) [inline e2e_tests/odh-minimal.yaml:48]: Same here

**Review by danielezonca — COMMENTED** [review]
Just a couple of comments about followup cleanups

File: e2e_tests/model.yaml (hunk 4 above)
Comment (ruivieira) [reply e2e_tests/model.yaml:13]: I'll add it to this PR

**Pull request merged**